### PR TITLE
feat: allow user to create opensearch alert

### PR DESCRIPTION
**Pull request by timckt**
This module allows user to create their OpenSearch Alert in Cloud Platform OpenSearch Cluster with terraform code.

Relates to ministryofjustice/cloud-platform#5476

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -2,42 +2,164 @@
...
-# Add module-specific variables here
-
 ########
 # Tags #
 ########
-variable "business_unit" {
-  description = "Area of the MOJ responsible for the service"
+
+# variable "business_unit" {
+#   description = "Area of the MOJ responsible for the service"
```
Comment (jaskaransarkaria) [inline variables.tf]: Can we remove any commented out please

File: README.md [hunk 2]
```diff
@@ -1,45 +1,73 @@
-# cloud-platform-terraform-template
+# cloud-platform-terraform-opensearch-alert
 
-[![Releases](https://img.shields.io/github/v/release/ministryofjustice/cloud-platform-terraform-template.svg)](https://github.com/ministryofjustice/cloud-platform-terraform-template/releases)
+[![Releases](https://img.shields.io/github/v/release/ministryofjustice/cloud-platform-terraform-opensearch-alert.svg)](https://github.com/ministryofjustice/cloud-platform-terraform-opensearch-alert/releases)
+
+This Terraform module creates an OpenSearch alert for detecting specific conditions and sends notifications to a Slack channel. It allows users to configure various aspects of the alert, including the query, triggers, and Slack channel settings.
```
Comment (jaskaransarkaria) [inline README.md]: Can you specify explicitly that this is only to create alerts for user application logs. It's also be worth creating a ticket, to add a feature in future to allow alerts to be created with this module for our modsec opensearch cluster

File: README.md [hunk 3]
```diff
@@ -1,45 +1,73 @@
...
-  infrastructure_support = var.infrastructure_support
+module "opensearch_alert" {
+  source = "github.com/ministryofjustice/cloud-platform-terraform-opensearch-alerts?ref=1.0.0" # use the latest module
+
+  secret_name                    = "secret-name-created-by-secret-manager-module"
+  secret_key                     = "url"
+  environment_name               = var.environment
+  slack_channel_name             = "your-slack-channel-name"
+  slack_channel_name_description = "slack-channel-description"
+  opensearch_alert_name          = "example-opensearch-alert-name"
```
Comment (poornima-krishnasamy) [inline README.md]: You can add a comment saying: To main uniqueness, the actual alert will be in the format of `"${var.environment_name}-${var.opensearch_alert_name}-${local.identifier}"`

File: README.md (hunk 2 above)
Comment (timckt) [reply README.md]: updated the readme. I think we can add the variable to the data block. 

```
data "aws_opensearch_domain" "live_app_logs" {
  domain_name = var.domain_name
}

data "aws_iam_role" "os_access_role_app_logs" {
  name = var.aws_iam_role
}
```

We can keep the default variable value to our user with `var.domain_name = "cp-live-app-logs" `and `var.aws_iam_role = "opensearch-access-role-app-logs"`

If we would like to use it for modsec opensearch cluster, we can change the variable respectively in our terraform code

**Comment by timckt** [comment]
Update the modules.
- add flexibility to re-use the module in other Cloud Platform OpenSearch cluster
- Add a github action to do a unit test.
- Update example folder with the correct usage
- Enable branch protection
- Added a small comment on the alert name

**Review by jaskaransarkaria — DISMISSED** [review]
LGTM

**Review by poornima-krishnasamy — APPROVED** [review]

**Pull request merged**